apigee_sync.go

```diff
@@ -517,3 +529,18 @@ func mapIsSubset(a map[string]bool, b map[string]bool) bool {
 
 	return true
 }
+
+/*
+ * Returns true if the two arrays have matching contents
+ */
+func ArrayEquals(a, b []string) bool {
```

Comment:
You don't need to write your own ArrayEqual, you could use reflect.deepequal instead.
FYI https://golang.org/pkg/reflect/